Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java

```diff
@@ -136,34 +136,58 @@ public void sessionDestroyed(IGameServerForMainServer gameServer, String roomNam
         LOGGER.log(Level.INFO, "Session {0} destroyed.", roomName);
     }
 
-    public void increaseSessionOccupancy(IGameServerForMainServer gameServer, SessionData sessionData) {
+    public void hostChanged(IGameServerForMainServer gameServer, String roomName, String newHostName) {
         synchronized (sessions) {
-            Optional<SessionData> session = sessions.get(gameServer).stream().filter(s -> s.getRoomName().equals(sessionData.getRoomName())).findFirst();
-            if (!session.isPresent()) {
+            List<SessionData> data = sessions.get(gameServer);
+
+            Optional<SessionData> sessionData = data.stream().filter(s -> s.getRoomName().equals(roomName)).findFirst();
+            if (!sessionData.isPresent()) {
+                return;
+            }
+
+            SessionData s = sessionData.get();
+            s.setHostName(newHostName);
+        }
+
+        LOGGER.log(Level.INFO, "Host of room {0} changed to {1}.", new Object[]{roomName, newHostName});
+    }
+
+    public void increaseSessionOccupancy(IGameServerForMainServer gameServer, String roomName) {
+        synchronized (sessions) {
+            List<SessionData> data = sessions.get(gameServer);
+
+            Optional<SessionData> sessionData = data.stream().filter(s -> s.getRoomName().equals(roomName)).findFirst();
+            if (!sessionData.isPresent()) {
                 return;
             }
 
-            SessionData s = session.get();
+            SessionData s = sessionData.get();
             int occupation = s.getOccupation();
             if (s.getCapacity() > occupation) {
                 s.setOccupation(occupation + 1);
             }
         }
+
+        LOGGER.log(Level.INFO, "Occupancy for room {0} increased.", roomName);
     }
 
-    public void decreaseSessionOccupancy(IGameServerForMainServer gameServer, SessionData sessionData) {
+    public void decreaseSessionOccupancy(IGameServerForMainServer gameServer, String roomName) {

```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)